### PR TITLE
Marketplace and Collection Stats

### PR DESCRIPTION
**Pull request by kespinola**
## Changes
- Query stats for marktplace and collections

### CODE REVIEW

File: crates/core/src/db/queries/stats.rs [hunk 1]
```diff
@@ -15,37 +15,43 @@ use crate::{
...
-            then pr.price
-        else 0
-    end)::bigint as volume_24hr
-
-from auction_houses ah
-    inner join purchase_receipts pr
-        on (pr.auction_house = ah.address)
+    auction_house,
+    mint,
+    min(listing_price) filter (where token_account_amount = 1 and listing_canceled_at is null and listing_purchase_receipt is null) as floor,
```
Comment (ray-kast) [inline crates/core/src/db/queries/stats.rs:25]:
```suggestion
    min(listing_price)::bigint filter (where token_account_amount = 1 and listing_canceled_at is null and listing_purchase_receipt is null) as floor,
```

File: crates/core/src/db/queries/stats.rs [hunk 2]
```diff
@@ -15,37 +15,43 @@ use crate::{
...
+    round(avg(purchase_price))::bigint as average,
+    sum(purchase_price) filter (where ($3 - purchased_at) < interval '24 hr') as volume_24hr
```
Comment (ray-kast) [inline crates/core/src/db/queries/stats.rs:27]:
```suggestion
    sum(purchase_price)::bigint filter (where ($3 - purchased_at) < interval '24 hr') as volume_24hr
```

File: crates/core/src/db/queries/stats.rs [hunk 3]
```diff
@@ -89,3 +94,59 @@ pub fn marketplace(
         .load(conn)
         .context("Failed to load marketplace stats")
 }
+
+const COLLECTION_QUERY: &str = r"
+select
+    auction_house,
+    mint,
+    min(listing_price) filter (where token_account_amount = 1 and listing_canceled_at is null and listing_purchase_receipt is null) as floor,
```
Comment (ray-kast) [inline crates/core/src/db/queries/stats.rs:102]:
```suggestion
    min(listing_price)::bigint filter (where token_account_amount = 1 and listing_canceled_at is null and listing_purchase_receipt is null) as floor,
```

File: crates/core/src/db/queries/stats.rs [hunk 4]
```diff
@@ -89,3 +94,59 @@ pub fn marketplace(
...
+    round(avg(purchase_price))::bigint as average,
+    sum(purchase_price) filter (where ($3 - purchased_at) < interval '24 hr') as volume_24hr
```
Comment (ray-kast) [inline crates/core/src/db/queries/stats.rs:104]:
```suggestion
    sum(purchase_price)::bigint filter (where ($3 - purchased_at) < interval '24 hr') as volume_24hr
```

**Review by ray-kast — APPROVED** [review]

File: crates/graphql/src/schema/objects/creator.rs [hunk 5]
```diff
@@ -22,13 +24,58 @@ struct AttributeGroup {
...
+    fn creations(&self, context: &AppContext) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+
+        let count = metadata_creators::table
+            .filter(metadata_creators::creator_address.eq(&self.creator.address))
+            .filter(metadata_creators::verified.eq(true))
+            .count()
+            .get_result::<i64>(&conn)?;
+
+        Ok(i64::try_into(count)?)
```
Comment (ray-kast) [inline crates/graphql/src/schema/objects/creator.rs:50]:
```suggestion
        Ok(count.try_into()?)
```

**Review by ray-kast — CHANGES_REQUESTED** [review]

File: crates/graphql/src/schema/objects/creator.rs [hunk 6]
```diff
@@ -22,13 +24,58 @@ struct AttributeGroup {
...
+    }
+
+    #[graphql(arguments(auction_houses(description = "Auction house public keys")))]
+    pub async fn stats(
+        &self,
+        auction_houses: Vec<PublicKey<AuctionHouse>>,
+        ctx: &AppContext,
+    ) -> FieldResult<Vec<MintStats>> {
+        let conn = ctx.db_pool.get()?;
+        let rows = stats::collection(&conn, auction_houses, &self.address)?;
```
Comment (ray-kast) [inline crates/graphql/src/schema/objects/creator.rs:71]: Should this be a data loader?

**Review by ray-kast — CHANGES_REQUESTED** [review]

File: crates/graphql/src/schema/objects/creator.rs (hunk 6 above)
Comment (kespinola) [reply crates/graphql/src/schema/objects/creator.rs:71]: Yes but we only load 1 creator at a time and its not a clean batch. need auction_houses to be fixed across the creators.

**Review by ray-kast — APPROVED** [review]

**Pull request merged**